### PR TITLE
Voting redux

### PR DESCRIPTION
**Pull request by stacy-wells**
ajax functionality is working

### CODE REVIEW

File: spec/features/reviews/user_votes_on_review_spec.rb [hunk 1]
```diff
@@ -22,37 +22,52 @@
...
+    scenario "user can't upvote more than once", js: true do
+      review = FactoryGirl.create(:review)
+
+      visit dinosaur_path(review.dinosaur)
+      page.find('.upvote-button').click
+      page.find('.upvote-button').click
+
+      expect(page).to have_content('1')
+    end
+
```
Comment (stacy-wells) [inline spec/features/reviews/user_votes_on_review_spec.rb]: We are leaving this space on purpose.

Comment (mlg-) [inline spec/features/reviews/user_votes_on_review_spec.rb]: Yeah, Hound isn't barking about the blank line, it's actually two spaces on line 62.

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -14,5 +14,41 @@
...
+    method: 'POST',
+    url: button.parent().attr('action'),
+    dataType: 'json',
+    data: {vote: {value: val}}
+  })
+  .success(function(data){
+    var score = $(button.parent().parent().children('.score'));
+    score.text(data);
+  });
+};
```
Comment (mlg-) [inline app/assets/javascripts/application.js]: It's not a great idea to put javascript directly into this application.js file. I would extract out the functionality you need into a separate js file.

File: app/models/review.rb [hunk 3]
```diff
@@ -2,12 +2,21 @@ class Review < ActiveRecord::Base
...
   validates :rating, presence: true
   validates :rating, inclusion: { in: 1..5, message: "Must be between 1 and 5" }
   validates :rating, numericality: { only_integer: true }
+
+  def tally
+    sum = 0
+    votes.each do |vote|
+      sum += vote[:value]
+    end
+    sum
```
Comment (mlg-) [inline app/models/review.rb]: You could use `reduce` here! `votes.reduce(:+)`.

**Pull request merged**